// ng4-material-dropdown/dist/src/modules/ng4-dropdown.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var ng4_dropdown_1 = require("./components/dropdown/ng4-dropdown");
exports.Ng4Dropdown = ng4_dropdown_1.Ng4Dropdown;
var ng4_dropdown_menu_1 = require("./components/menu/ng4-dropdown-menu");
exports.Ng4DropdownMenu = ng4_dropdown_menu_1.Ng4DropdownMenu;
var ng4_dropdown_button_1 = require("./components/button/ng4-dropdown-button");
exports.Ng4DropdownButton = ng4_dropdown_button_1.Ng4DropdownButton;
var ng4_menu_item_1 = require("./components/menu-item/ng4-menu-item");
exports.Ng4MenuItem = ng4_menu_item_1.Ng4MenuItem;
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var animations_1 = require("@angular/platform-browser/animations");
var dropdown_state_service_1 = require("./services/dropdown-state.service");
exports.DropdownStateService = dropdown_state_service_1.DropdownStateService;
var Ng4DropdownModule = (function () {
    function Ng4DropdownModule() {
    }
    return Ng4DropdownModule;
}());
Ng4DropdownModule = __decorate([
    core_1.NgModule({
        exports: [
            ng4_menu_item_1.Ng4MenuItem,
            ng4_dropdown_button_1.Ng4DropdownButton,
            ng4_dropdown_menu_1.Ng4DropdownMenu,
            ng4_dropdown_1.Ng4Dropdown
        ],
        declarations: [
            ng4_dropdown_1.Ng4Dropdown,
            ng4_menu_item_1.Ng4MenuItem,
            ng4_dropdown_button_1.Ng4DropdownButton,
            ng4_dropdown_menu_1.Ng4DropdownMenu,
        ],
        imports: [common_1.CommonModule, animations_1.BrowserAnimationsModule]
    })
], Ng4DropdownModule);
exports.Ng4DropdownModule = Ng4DropdownModule;
