// ng4-material-dropdown/dist/src/modules/components/menu-item/ng4-menu-item.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var dropdown_state_service_1 = require("../../services/dropdown-state.service");
var Ng4MenuItem = (function () {
    function Ng4MenuItem(state, element, renderer) {
        this.state = state;
        this.element = element;
        this.renderer = renderer;
        this.preventClose = false;
    }
    Object.defineProperty(Ng4MenuItem.prototype, "isSelected", {
        get: function () {
            return this === this.state.dropdownState.selectedItem;
        },
        enumerable: true,
        configurable: true
    });
    Ng4MenuItem.prototype.select = function ($event) {
        this.state.dropdownState.select(this, true);
        if ($event) {
            $event.stopPropagation();
            $event.preventDefault();
        }
    };
    Ng4MenuItem.prototype.click = function () {
        this.state.dropdownState.onItemClicked.emit(this);
    };
    Ng4MenuItem.prototype.focus = function () {
        this.renderer.invokeElementMethod(this.element.nativeElement.children[0], 'focus');
    };
    return Ng4MenuItem;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], Ng4MenuItem.prototype, "preventClose", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], Ng4MenuItem.prototype, "value", void 0);
Ng4MenuItem = __decorate([
    core_1.Component({
        selector: 'ng4-menu-item',
        styleUrls: ['./style.scss'],
        templateUrl: './template.html'
    }),
    __metadata("design:paramtypes", [dropdown_state_service_1.DropdownStateService,
        core_1.ElementRef,
        core_1.Renderer])
], Ng4MenuItem);
exports.Ng4MenuItem = Ng4MenuItem;
