// ng4-material-dropdown/dist/components/menu-item/ng4-menu-item.js
import { Component, Input, Renderer, ElementRef } from '@angular/core';
import { DropdownStateService } from '../../services/dropdown-state.service';
var Ng4MenuItem = (function () {
    function Ng4MenuItem(state, element, renderer) {
        this.state = state;
        this.element = element;
        this.renderer = renderer;
        this.preventClose = false;
    }
    Object.defineProperty(Ng4MenuItem.prototype, "isSelected", {
        get: function () {
            return this === this.state.dropdownState.selectedItem;
        },
        enumerable: true,
        configurable: true
    });
    Ng4MenuItem.prototype.select = function ($event) {
        this.state.dropdownState.select(this, true);
        if ($event) {
            $event.stopPropagation();
            $event.preventDefault();
        }
    };
    Ng4MenuItem.prototype.click = function () {
        this.state.dropdownState.onItemClicked.emit(this);
    };
    Ng4MenuItem.prototype.focus = function () {
        this.renderer.invokeElementMethod(this.element.nativeElement.children[0], 'focus');
    };
    return Ng4MenuItem;
}());
export { Ng4MenuItem };
Ng4MenuItem.decorators = [
    { type: Component, args: [{
                selector: 'ng4-menu-item',
                styleUrls: ['./style.scss'],
                templateUrl: './template.html'
            },] },
];
Ng4MenuItem.ctorParameters = function () { return [
    { type: DropdownStateService, },
    { type: ElementRef, },
    { type: Renderer, },
]; };
Ng4MenuItem.propDecorators = {
    'preventClose': [{ type: Input },],
    'value': [{ type: Input },],
};
