// ng4-material-dropdown/dist/components/button/ng4-dropdown-button.js
import { Component, Output, Input, EventEmitter, ElementRef } from '@angular/core';
var Ng4DropdownButton = (function () {
    function Ng4DropdownButton(element) {
        this.element = element;
        this.onMenuToggled = new EventEmitter();
        this.showCaret = true;
    }
    Ng4DropdownButton.prototype.toggleMenu = function () {
        this.onMenuToggled.emit(true);
    };
    Ng4DropdownButton.prototype.getPosition = function () {
        return this.element.nativeElement.getBoundingClientRect();
    };
    return Ng4DropdownButton;
}());
export { Ng4DropdownButton };
Ng4DropdownButton.decorators = [
    { type: Component, args: [{
                selector: 'ng4-dropdown-button',
                styleUrls: ['./style.scss'],
                templateUrl: './template.html'
            },] },
];
Ng4DropdownButton.ctorParameters = function () { return [
    { type: ElementRef, },
]; };
Ng4DropdownButton.propDecorators = {
    'onMenuToggled': [{ type: Output },],
    'showCaret': [{ type: Input },],
};
