// ng4-material-dropdown/dist/components/dropdown/ng4-dropdown.js
import { HostListener, Component, ContentChild, Output, EventEmitter, Input } from '@angular/core';
import { Ng4DropdownButton } from '../button/ng4-dropdown-button';
import { Ng4DropdownMenu } from '../menu/ng4-dropdown-menu';
import { DropdownStateService } from '../../services/dropdown-state.service';
var Ng4Dropdown = (function () {
    function Ng4Dropdown(state) {
        this.state = state;
        this.dynamicUpdate = true;
        this.onItemClicked = new EventEmitter();
        this.onItemSelected = new EventEmitter();
        this.onShow = new EventEmitter();
        this.onHide = new EventEmitter();
    }
    Ng4Dropdown.prototype.toggleMenu = function (position) {
        if (position === void 0) { position = this.button.getPosition(); }
        this.state.menuState.isVisible ? this.hide() : this.show(position);
    };
    Ng4Dropdown.prototype.hide = function () {
        this.menu.hide();
        this.onHide.emit(this);
    };
    Ng4Dropdown.prototype.show = function (position) {
        if (position === void 0) { position = this.button.getPosition(); }
        this.menu.show();
        this.menu.updatePosition(position);
        this.onShow.emit(this);
    };
    Ng4Dropdown.prototype.scrollListener = function () {
        if (this.state.menuState.isVisible && this.button && this.dynamicUpdate) {
            this.menu.updatePosition(this.button.getPosition());
        }
    };
    Ng4Dropdown.prototype.ngOnInit = function () {
        var _this = this;
        this.state.dropdownState.onItemClicked.subscribe(function (item) {
            _this.onItemClicked.emit(item);
            if (item.preventClose) {
                return;
            }
            _this.hide.call(_this);
        });
        if (this.button) {
            this.button.onMenuToggled.subscribe(function () {
                _this.toggleMenu();
            });
        }
        this.state.dropdownState.onItemSelected.subscribe(function (item) { return _this.onItemSelected.emit(item); });
    };
    return Ng4Dropdown;
}());
export { Ng4Dropdown };
Ng4Dropdown.decorators = [
    { type: Component, args: [{
                selector: 'ng4-dropdown',
                templateUrl: './template.html',
                providers: [DropdownStateService]
            },] },
];
Ng4Dropdown.ctorParameters = function () { return [
    { type: DropdownStateService, },
]; };
Ng4Dropdown.propDecorators = {
    'button': [{ type: ContentChild, args: [Ng4DropdownButton,] },],
    'menu': [{ type: ContentChild, args: [Ng4DropdownMenu,] },],
    'dynamicUpdate': [{ type: Input },],
    'onItemClicked': [{ type: Output },],
    'onItemSelected': [{ type: Output },],
    'onShow': [{ type: Output },],
    'onHide': [{ type: Output },],
    'scrollListener': [{ type: HostListener, args: ['window:scroll',] },],
};
