// ng4-material-dropdown/dist/src/modules/components/dropdown/ng4-dropdown.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var ng4_dropdown_button_1 = require("../button/ng4-dropdown-button");
var ng4_dropdown_menu_1 = require("../menu/ng4-dropdown-menu");
var dropdown_state_service_1 = require("../../services/dropdown-state.service");
var Ng4Dropdown = (function () {
    function Ng4Dropdown(state) {
        this.state = state;
        this.dynamicUpdate = true;
        this.onItemClicked = new core_1.EventEmitter();
        this.onItemSelected = new core_1.EventEmitter();
        this.onShow = new core_1.EventEmitter();
        this.onHide = new core_1.EventEmitter();
    }
    Ng4Dropdown.prototype.toggleMenu = function (position) {
        if (position === void 0) { position = this.button.getPosition(); }
        this.state.menuState.isVisible ? this.hide() : this.show(position);
    };
    Ng4Dropdown.prototype.hide = function () {
        this.menu.hide();
        this.onHide.emit(this);
    };
    Ng4Dropdown.prototype.show = function (position) {
        if (position === void 0) { position = this.button.getPosition(); }
        this.menu.show();
        this.menu.updatePosition(position);
        this.onShow.emit(this);
    };
    Ng4Dropdown.prototype.scrollListener = function () {
        if (this.state.menuState.isVisible && this.button && this.dynamicUpdate) {
            this.menu.updatePosition(this.button.getPosition());
        }
    };
    Ng4Dropdown.prototype.ngOnInit = function () {
        var _this = this;
        this.state.dropdownState.onItemClicked.subscribe(function (item) {
            _this.onItemClicked.emit(item);
            if (item.preventClose) {
                return;
            }
            _this.hide.call(_this);
        });
        if (this.button) {
            this.button.onMenuToggled.subscribe(function () {
                _this.toggleMenu();
            });
        }
        this.state.dropdownState.onItemSelected.subscribe(function (item) { return _this.onItemSelected.emit(item); });
    };
    return Ng4Dropdown;
}());
__decorate([
    core_1.ContentChild(ng4_dropdown_button_1.Ng4DropdownButton),
    __metadata("design:type", ng4_dropdown_button_1.Ng4DropdownButton)
], Ng4Dropdown.prototype, "button", void 0);
__decorate([
    core_1.ContentChild(ng4_dropdown_menu_1.Ng4DropdownMenu),
    __metadata("design:type", ng4_dropdown_menu_1.Ng4DropdownMenu)
], Ng4Dropdown.prototype, "menu", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Object)
], Ng4Dropdown.prototype, "dynamicUpdate", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], Ng4Dropdown.prototype, "onItemClicked", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], Ng4Dropdown.prototype, "onItemSelected", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], Ng4Dropdown.prototype, "onShow", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], Ng4Dropdown.prototype, "onHide", void 0);
__decorate([
    core_1.HostListener('window:scroll'),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], Ng4Dropdown.prototype, "scrollListener", null);
Ng4Dropdown = __decorate([
    core_1.Component({
        selector: 'ng4-dropdown',
        templateUrl: './template.html',
        providers: [dropdown_state_service_1.DropdownStateService]
    }),
    __metadata("design:paramtypes", [dropdown_state_service_1.DropdownStateService])
], Ng4Dropdown);
exports.Ng4Dropdown = Ng4Dropdown;
