// ng4-material-dropdown/dist/src/modules/components/dropdown/test-helpers.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var ng4_dropdown_module_1 = require("../../ng4-dropdown.module");
var BasicDropdown = (function () {
    function BasicDropdown() {
    }
    BasicDropdown.prototype.ngOnInit = function () { };
    return BasicDropdown;
}());
BasicDropdown = __decorate([
    core_1.Component({
        selector: 'basic-dropdown',
        template: "\n        <main>\n        <ng4-dropdown>\n            <ng4-dropdown-button>\n                Open\n            </ng4-dropdown-button>\n            <ng4-dropdown-menu [focusFirstElement]=\"true\">\n                <ng4-menu-item>\n                    First item\n                </ng4-menu-item>\n                <ng4-menu-item [preventClose]=\"true\">\n                    Second item\n                </ng4-menu-item>\n            </ng4-dropdown-menu>\n        </ng4-dropdown>\n        </main>\n    "
    })
], BasicDropdown);
exports.BasicDropdown = BasicDropdown;
var TestModule = (function () {
    function TestModule() {
    }
    return TestModule;
}());
TestModule = __decorate([
    core_1.NgModule({
        declarations: [
            BasicDropdown
        ],
        imports: [ng4_dropdown_module_1.Ng4DropdownModule],
        exports: [BasicDropdown]
    })
], TestModule);
exports.TestModule = TestModule;
